google/cloud/spanner_v1/batch.py

```diff
@@ -217,10 +217,7 @@ def commit(
                 request=request,
                 metadata=metadata,
             )
-            response = _retry(
-                method,
-                allowed_exceptions={InternalServerError: _check_rst_stream_error},
-            )
+            response = self._session.run_in_transaction(method)
```

Comment:
Hmmm.... Now that I see this, I realize that this cannot use `run_in_transaction` directly. The reason is that:
1. This method `batch.commit()` in its original form (so without the changes in this pull request) creates and commits a single-use read/write transaction.
2. `run_in_transaction` however creates a new read/write transaction and then executes the given method in the scope of that transaction.

So what happens now is that you create two transactions:
1. `run_in_transaction` creates a transaction that is not being used.
2. `method` that is passed in to `run_in_transaction` executes a `Commit` in a single-use read/write transaction.